### PR TITLE
Docker, installation and documentation improvements 

### PR DESCRIPTION
**Pull request by passalis**
This PR improves docker images and installation process by
- Removing cache from docker images (about ~3GB size reduction)
- Reducing the number of intermediate layers created
- Adding missing ROS packages after building the main dependencies
- Adding a default argument (master) for branch creation
- Transferred docker building process into a separate customization section of the documentation

### CODE REVIEW

File: docs/reference/installation.md [hunk 1]
```diff
@@ -1,68 +1,29 @@
...
-| pip                 | Both     | Ubuntu 20.04 (x86-64) |
-| docker              | Both     | Linux / Windows       |
+| Installation Method   | OS                    |
+|-----------------------|-----------------------|
+| Clone & Install       | Ubuntu 20.04 (x86-64) |
+| pip                   | Ubuntu 20.04 (x86-64) |
+| docker                | Linux / Windows       |
 
+Note that pip installation includes only the Python API of the toolkit. 
+If you need to use all the functionalities of the toolkit (e.g., ROS nodes, etc.), then you need either to use the pre-compiled docker images or to follow the installation instruction for cloning and building the toolkit.
```
Comment (stefaniapedrazzi) [inline docs/reference/installation.md:17]:
```suggestion
If you need to use all the functionalities of the toolkit (e.g., ROS nodes, etc.), then you need either to use the pre-compiled docker images or to follow the installation instructions for cloning and building the toolkit.
```

File: docs/reference/installation.md [hunk 2]
```diff
@@ -1,68 +1,29 @@
...
-make ctests
-```
-
-If you plan to use GPU-enabled functionalities, then you are advised to install [CUDA 11.2](https://developer.nvidia.com/cuda-11.2.0-download-archive), along with [CuDNN](https://developer.nvidia.com/cudnn).
-
-**HINT:** All tests probe for the `TEST_DEVICE` enviromental variable when running.
-If this enviromental variable is set during testing, it allows for easily running all tests on a different device (e.g., setting `TEST_DEVICE=cuda:0` runs all tests on the first GPU of the system).
+before installing the toolkit and then follow the installation instructions in the relevant section.
+All the required dependecies will be automatically installed (or explicit instructions are provided).
+Other platforms apart from Ubuntu 20.04, e.g., Windows, other linux distributions, etc., are currently supported through docker images.
```
Comment (stefaniapedrazzi) [inline docs/reference/installation.md:26]:
```suggestion
Other platforms apart from Ubuntu 20.04, e.g., Windows, other Linux distributions, etc., are currently supported through docker images.
```

**Review by stefaniapedrazzi — COMMENTED** [review]
Thank you!

I just have two very minor suggestions to improve the text.

File: docs/reference/installation.md [hunk 3]
```diff
@@ -1,68 +1,29 @@
...
-make unittest
-make ctests
-```
-
-If you plan to use GPU-enabled functionalities, then you are advised to install [CUDA 11.2](https://developer.nvidia.com/cuda-11.2.0-download-archive), along with [CuDNN](https://developer.nvidia.com/cudnn).
-
-**HINT:** All tests probe for the `TEST_DEVICE` enviromental variable when running.
-If this enviromental variable is set during testing, it allows for easily running all tests on a different device (e.g., setting `TEST_DEVICE=cuda:0` runs all tests on the first GPU of the system).
+before installing the toolkit and then follow the installation instructions in the relevant section.
+All the required dependecies will be automatically installed (or explicit instructions are provided).
```
Comment (ad-daniel) [inline docs/reference/installation.md:25]:
```suggestion
All the required dependencies will be automatically installed (or explicit instructions are provided).
```

**Review by ad-daniel — CHANGES_REQUESTED** [review]
Thank you

**Review by ad-daniel — APPROVED** [review]

**Pull request merged**